libraries/pbrlib/genglsl/lib/mx_microfacet_diffuse.glsl

```diff
@@ -79,6 +79,33 @@ float mx_oren_nayar_diffuse_dir_albedo(float NdotV, float roughness)
     return clamp(dirAlbedo, 0.0, 1.0);
 }
 
+float mx_oren_nayar_diffuse_avg_albedo(float NdotV, float roughness)
+{
+    float sigma2 = mx_square(roughness);
+    float A = 1.0 - 0.5 * (sigma2 / (sigma2 + 0.33));
+    float B = 0.45 * sigma2 / (sigma2 + 0.09);
+    return A + (2.0 / 3.0 - 64.0 / (45.0 * M_PI)) * B;
```

Comment:
This is what we called the QON model in the paper, not FON.
If you compensate this, you don't get what we called EON (which is compensated FON). 

For example, it will have the dark circle artifacts, and (subjectively) doesn't look quite as nice as EON.

I'd suggest that the "compensated Oren-Nayar" should actually modify the QON term to FON, as in the paper.